### PR TITLE
Call subquery

### PR DESCRIPTION
**Pull request by raz-mon**
This PR adds support for the CALL {} clause.

### CODE REVIEW

File: tests/flow/test_call_subquery.py [hunk 1]
```diff
@@ -0,0 +1,892 @@
...
+                "WITH imports in CALL {} must be simple ('WITH a')")
+
+        # non-valid queries within CALL {}
+        for query in [
+            "CALL {CREATE (n:N) MATCH (n:N) RETURN n} RETURN 1",
+            "WITH 1 AS a CALL {WITH a CREATE (n:N) MATCH (n:N) RETURN n} RETURN a",
+            "CALL {MATCH (n:N) CREATE (n:N2)} RETURN 1"
+        ]:
+            # just pass in case of an error, fail otherwise
+            self.expect_error(query, "")
```
Comment (AviAvni) [inline tests/flow/test_call_subquery.py:59]: split and specify error

File: tests/flow/test_call_subquery.py [hunk 2]
```diff
@@ -0,0 +1,892 @@
...
+            """
+            WITH a
+            CALL {
+                WITH a 
+                RETURN a AS b
+            } 
+            RETURN b
+            """,
+        ]:
+            self.expect_error(query, "a not defined")
```
Comment (AviAvni) [inline tests/flow/test_call_subquery.py:79]: add '<var>'

File: tests/flow/test_call_subquery.py [hunk 3]
```diff
@@ -0,0 +1,892 @@
...
+                RETURN x + n.v
+                """
+        self.expect_error(query, "x not defined")
+
+        # outer scope variables (bound) should not be returnable from the sq
+        query = "MATCH (n:N) CALL {RETURN 1 AS n} RETURN n"
+        self.expect_error(query, "Variable `n` already declared in outer scope")
+
+        # a CALL {} after an updating clause requires a separating WITH
+        query = "CREATE (n:N) CALL {RETURN 1} RETURN 1"
```
Comment (AviAvni) [inline tests/flow/test_call_subquery.py:133]: is this ok?
```
CREATE (n:N) CALL { CREATE (m:M) RETURN 1 } RETURN 1
```

**Review by AviAvni — CHANGES_REQUESTED** [review]

**Comment by filipecosta90** [comment]
### Automated performance analysis summary

This comment was automatically generated given there is performance data available.

In summary:
- Detected a total of 11 stable tests between versions.

You can check a comparison in detail via the [grafana link](https://benchmarksrediscom.grafana.net/d//SH9_rQYGz/?var-branch=master&var-branch=call_subquery)

### Comparison between master and call_subquery.

Time Period from 30 days ago. (environment used: oss-standalone)

|                               Test Case                               |Baseline master (median obs. +- std.dev)|Comparison call_subquery (median obs. +- std.dev)|% change (higher-better)|     Note      |
|-----------------------------------------------------------------------|----------------------------------------|-------------------------------------------------|------------------------|---------------|
|ENTITY_COUNT                                                           | 49998 +- 0.0%  (3 datapoints)          | 49999 +- nan%  (1 datapoints)                   |0.0%                    |-- no change --|
|EXPLICIT-EDGE-DELETION                                                 | 100 +- 0.0%  (3 datapoints)            | 100 +- nan%  (1 datapoints)                     |-0.0%                   |-- no change --|
|GRAPH500-SCALE_18-EF_16-1_HOP                                          | 33324 +- 0.0%  (3 datapoints)          | 33323 +- nan%  (1 datapoints)                   |-0.0%                   |-- no change --|
|GRAPH500-SCALE_18-EF_16-1_HOP_MIXED_READ_65perc_WRITE_25perc_DEL_10perc| 50 +- 0.0%  (3 datapoints)             | 50 +- nan%  (1 datapoints)                      |0.0%                    |-- no change --|
|GRAPH500-SCALE_18-EF_16-2_HOP_MIXED_READ_75perc_WRITE_25perc           | 59 +- 3.2%  (3 datapoints)             | 59 +- nan%  (1 datapoints)                      |0.0%                    |-- no change --|
|IMPLICIT-EDGE-DELETION                                                 | 17 +- 0.0%  (3 datapoints)             | 17 +- nan%  (1 datapoints)                      |-0.0%                   |-- no change --|
|INSPECT-EDGES                                                          | 2500 +- 0.0%  (3 datapoints)           | 2500 +- nan%  (1 datapoints)                    |-0.0%                   |-- no change --|
|MIX_READ_WRITE                                                         | 10000 +- 0.0%  (3 datapoints)          | 10000 +- nan%  (1 datapoints)                   |0.0%                    |-- no change --|
|NODE-INDEX-LOOKUP                                                      | 33 +- 0.0%  (3 datapoints)             | 33 +- nan%  (1 datapoints)                      |0.0%                    |-- no change --|
|SORT_ENTITIES                                                          | 100 +- 0.0%  (3 datapoints)            | 100 +- nan%  (1 datapoints)                     |-0.0%                   |-- no change --|
|allShortestPaths-10hop-100Kpaths                                       | 30 +- 0.0%  (3 datapoints)             | 30 +- nan%  (1 datapoints)                      |-0.0%                   |-- no change --|

**Pull request merged**